Add reviews to patch view and extend comment count to all revisions

sebastinez opened the pull request:
- Show comments in PatchTeaser over all revisions, instead of just the latest one.
- Add reviewers to patch sidebar

**Accept**
<img width="254" alt="Bildschirmfoto 2023-06-23 um 12 00 48" src="https://github.com/radicle-dev/radicle-interface/assets/7912302/75af296c-ace7-49c1-989b-674a6b73c540">

**Reject**
<img width="258" alt="Bildschirmfoto 2023-06-23 um 12 00 30" src="https://github.com/radicle-dev/radicle-interface/assets/7912302/f2e35508-3daa-4b61-8ea4-bd1ccab4fd14">

**Comment (on a superseded revision)**
<img width="254" alt="Bildschirmfoto 2023-06-23 um 12 00 13" src="https://github.com/radicle-dev/radicle-interface/assets/7912302/ed7e21c3-162e-4b6b-8ee9-4c12a9310429">


Closes #700

cloudhead commented:
Wow cool!

cloudhead commented on src/views/projects/Patch.svelte:
```diff
@@ -425,6 +450,27 @@
+              class:review-reject={review.verdict === "reject"}>
+              <Icon
+                size="small"
+                name={review.verdict === "accept" ? "checkmark" : "cross"} />
```
Sometimes there's no verdict, just a comment. Does this take it into account?

sebastinez replied:
Yeah haven't pushed the latest commit yet, I check for `accept` or `reject` verdict and else show the chat icon

sebastinez commented on src/components/Icon.svelte:
```diff
@@ -174,6 +175,11 @@
          14.3536C16.1583 14.5488 15.8417 14.5488 15.6464 14.3536L12.3536
          11.0607C12.1583 10.8654 11.8417 10.8654 11.6464 11.0607L8.35355
          14.3536C8.15829 14.5488 7.84171 14.5488 7.64645 14.3536Z" />
+  {:else if name === "cross"}
```
Thanks @rudolfs! Not sure we need the `cross-small` currently.. maybe we can add it later?

rudolfs replied:
Sure.

sebastinez replied:
Ok added in d3c80c5

sebastinez commented:
Okay addressed the comments that originated in #700, but didn't touch the pending `reviewers` string array for now, since we don't have yet a way to populate it.
Will address it once we can display something.

rudolfs commented on src/views/projects/Patch/PatchTeaser.svelte:
```diff
@@ -25,6 +25,12 @@
   );
+
+  // Counts the amount of comments in all the discussions over all revisions.
+  $: commentCount = patch.revisions.reduce(
```
Is this actually working, or do we have different kinds of comments? Because I don't see the comment counter in the list view.

http://localhost:3000/seeds/seed.radicle.xyz/rad:z3gqcJUoA1n9HaHKufZs5FCSGazv5/patches/67f8e36cf5f2ecb4635e0b24edf5aefa5dcb4099

<img width="1840" alt="Screenshot 2023-06-27 at 14 29 06" src="https://github.com/radicle-dev/radicle-interface/assets/158411/65f3b54c-0b7e-49d4-82c5-c31cce43272b">

<img width="1840" alt="Screenshot 2023-06-27 at 14 29 10" src="https://github.com/radicle-dev/radicle-interface/assets/158411/7ad34741-01b3-460b-901b-3214381679c6">

sebastinez replied:
We should probably rephrase a review without a verdict, maybe `left a review`?

rudolfs replied:
Yeah, changing `commented` to `left a review` sounds good.

sebastinez replied:
okay did that with commit 2a3e1a8

rudolfs commented:
Could we have a visual spec of the patch view for each of the review states?

 - accept
 - reject
 - comment

sebastinez commented:
> Could we have a visual spec of the patch view (so we have the sidebar in the visual diff) for each of the review states?
> 
> * accept
> * reject
> * comment

Added visual tests for all states.
![patch-page-3-visual-darwin](https://github.com/radicle-dev/radicle-interface/assets/7912302/730a9f88-6984-4769-a3b2-5997b7b9119b)
![patch-page-4-visual-darwin](https://github.com/radicle-dev/radicle-interface/assets/7912302/198e1e28-dcba-4d0a-a5dc-8d87bbc23e8c)
![patch-page-5-visual-darwin](https://github.com/radicle-dev/radicle-interface/assets/7912302/c0c3bedd-3378-4f31-92dd-f8ef11dd48f1)
